Tests/Unit/MSFT_xIISFeatureDelegation.tests.ps1

```diff
@@ -4,14 +4,49 @@ $global:DSCResourceName = 'MSFT_xIISFeatureDelegation'
 #region HEADER
 
 [String] $moduleRoot = Split-Path -Parent (Split-Path -Parent (Split-Path -Parent $Script:MyInvocation.MyCommand.Path))
- if ( (-not (Test-Path -Path (Join-Path -Path $moduleRoot -ChildPath 'DSCResource.Tests'))) -or `
-      (-not (Test-Path -Path (Join-Path -Path $moduleRoot -ChildPath 'DSCResource.Tests\TestHelper.psm1'))) )
+$repoSource = (Get-Module -Name $global:DSCModuleName -ListAvailable)
+
+if (($repoSource -ne $null) -and ($repoSource[0].RepositorySourceLocation.Host -eq 'www.powershellgallery.com'))
+{
+    if ( -not (Test-Path -Path (Join-Path -Path $moduleRoot -ChildPath 'Tests\DscResourceTestHelper')) )
+    {
+
+        $choice = read-host "In order to run this test you need to install a helper module, continue with installation? (Y/N)"
+
+        if ($choice -eq "y"){
+            #install test folders from gallery
+            Save-Module -Name 'DscResourceTestHelper' -Path (Join-Path -Path $moduleRoot -ChildPath 'Tests')
+        }
+    
+        else 
+        {
+            Write-Host "Unable to run tests without the required helper module - Exiting test"
+            return
+        }
+        
+    }
+
+    $testModule = Get-Module -Name 'DscResourceTestHelper' -ListAvailable
+    $testModuleVer = $testModule[0].Version
+    Import-Module (Join-Path -Path $moduleRoot -ChildPath "Tests\DscResourceTestHelper\$testModuleVer\TestHelper.psm1") -Force
+} 
+else
 {
-    & git @('clone','https://github.com/PowerShell/DscResource.Tests.git',(Join-Path -Path $moduleRoot -ChildPath '\DSCResource.Tests\'))
+    if (-not (Test-Path -Path (Join-Path -Path $moduleRoot -ChildPath '\Tests\DscResource.Tests\DscResourceTestHelper\TestHelper.psm1')))
+    {
+        #clone test folders from gitHub
+        $dscResourceTestsPath = Join-Path -Path $moduleRoot -ChildPath '\Tests\DscResource.Tests'
+        & git @('clone','https://github.com/PowerShell/DscResource.Tests.git',$dscResourceTestsPath)
+        Push-Location
+        cd $dscResourceTestsPath
+        & git checkout gitDependency
+        Pop-Location
+    }
+
+    Import-Module (Join-Path -Path $moduleRoot -ChildPath '\Tests\DscResource.Tests\DscResourceTestHelper\TestHelper.psm1') -Force
```

Comment:
You can store (Join-Path -Path $moduleRoot -ChildPath '\Tests\DscResource.Tests\DscResourceTestHelper\TestHelper.psm1') in separate variable and reuse to make it cleaner